### DISCUSSION TITLE
Pixel Shift Causing Discontinuities in Linear Features in Titiler

### DISCUSSION THREAD

**Discussion by pgulla** [developmentseed/titiler#733] [Q&A] [answered]
In the recent version of Titiler, it has been observed that at higher zoom levels, linear features exhibit discontinuities due to a one-pixel shift in the tiles. This issue is not present in the older version of Titiler (0.11.5). Attached are screenshots of the same image in both the older (0.11.5) and newer (0.15.6) versions of Titiler for comparison.

Please review the provided screenshots for a visual comparison.

Titiler 0.11.5

![Pasted Graphic](https://github.com/developmentseed/titiler/assets/105771604/d2df2f57-2828-4c35-a077-b3d76920d80d)

Titiler 0.15.6

![Pasted Graphic 1](https://github.com/developmentseed/titiler/assets/105771604/83f747fc-91bb-4417-8d4d-674a994db828)

The stac Item: https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l2a-cogs/items/S2B_10SEJ_20180927_0_L2A

**Comment by vincentsarago** [comment]
@pgulla can you provide a bit more info about infrastructure and also the GDAL version (`rio --show-versions`)


~~I've created a really simple dataset to try to reproduce this but I can't see the shift~~

```
https://titiler.xyz/cog/map?url=s3://ds-satellite/cogs/cog_binary_UTM.tif&reproject=bilinear
```


<img width="2565" alt="Screenshot 2023-11-30 at 11 09 17 AM" src="https://github.com/developmentseed/titiler/assets/10407788/bab647a8-b62c-4a61-add8-635c49277818">


**edited**: updated the dataset to use UTM projection to match the Sentinel2 Dataset

**Comment by vincentsarago** [comment, marked as answer]
Ok looking at the data I was able to reproduce and fix it by using a different `reprojection` resampling method (`reproject=cubic|bilinear`)

`reproject=nearest` (default)

![](https://github.com/developmentseed/titiler/assets/10407788/34f7658d-3757-4769-8483-1e733a6e53ad)

https://titiler.xyz/stac/map?url=https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l2a-cogs/items/S2B_10SEJ_20180927_0_L2A&assets=B04&assets=B03&assets=B02&assets_as_band=True&rescale=0,4000&minzoom=8&maxzoom=14&reproject=nearest

`reproject=bilinear` (default)

![](https://github.com/developmentseed/titiler/assets/10407788/a1de90b9-6474-405a-be5d-0246e706c807)

https://titiler.xyz/stac/map?url=https://earth-search.aws.element84.com/v0/collections/sentinel-s2-l2a-cogs/items/S2B_10SEJ_20180927_0_L2A&assets=B04&assets=B03&assets=B02&assets_as_band=True&rescale=0,4000&minzoom=8&maxzoom=14&reproject=bilinear
